code/log/lib/Loot/Log/Component.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE RecordWildCards #-}
+module Loot.Log.Component
+       ( allocateLogging
+       , prepareLogging
+       ) where
+
+import Loot.Log.Actions
+import Loot.Log.Config
+import Loot.Log.Internal
+
+import Colog.Core.Action (LogAction (..), cfilter)
+import Colog.Syslog (mkSyslogHandler, shClose)
+import Control.Monad.Component (ComponentM, buildComponent)
+import System.IO (hClose)
+
+-- | Builds a 'Logging' 'ComponentM' with the name "logging" from a 'LogConfig'
+allocateLogging
+    :: MonadIO m
+    => LogConfig
+    -> NameSelector
+    -> ComponentM (Logging m)
+allocateLogging config nameSel = do
+    (logging, clean) <- liftIO $ prepareLogging config nameSel
+    buildComponent "logging" (pure logging) (const clean)
```

Comment:
That is an incorrect use of `buildComponent`. See, you perform actual allocatiion action _outside_ of `buildComponent` bracket, therefore if an exception occurs while allocating some part of `Logging`, the cleanup action will not be called and the error reported by `ComponentM` will not mention `"logging"` component, because this error will be thrown outside `buildComponent "logging"` bracket.
This can easily lead to leaking resources: e. g. if a file backend got allocated successfully but an error occured during allocating syslog backend, the handle for the file backend will not be closed.
The more proper way of allocating logging backends would be allocating each backend in its own `buildComponent` (or `buildComponent_` for stderr/stdout). `allocateLogging` will be then the `ComponentM` action which builds a _bunch_ of components instead of just one.